File: timerx-example/src/main/java/com/arsvechkarev/timerxexample/MainActivity.java
package com.arsvechkarev.timerxexample;

import android.os.Bundle;
import android.view.View;
import android.view.View.OnClickListener;
import android.widget.TextView;
import android.widget.Toast;
import androidx.appcompat.app.AppCompatActivity;
import com.arsvechkarev.timerx.Stopwatch;
import com.arsvechkarev.timerx.TimeTickListener;
import com.arsvechkarev.timerx.TimeUnits;

public class MainActivity extends AppCompatActivity {

  private TextView textTimer;
  private Stopwatch stopwatch;

  @Override
  protected void onCreate(final Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setContentView(R.layout.activity_main);

    textTimer = findViewById(R.id.text_timer);
    stopwatch = new Stopwatch(new TimeTickListener() {
      @Override
      public void onTimeTick(String time) {
        textTimer.setText(time);
      }
    }, "Format#Start: MM::SS::")
        .changeFormatWhen(10, TimeUnits.SECONDS, "Format10: MM:SS")
        .changeFormatWhen(5, TimeUnits.SECONDS, "Format5: SSSSSS")
        .changeFormatWhen(15, TimeUnits.SECONDS, "Format15: SS#L");

    findViewById(R.id.btn_start).setOnClickListener(new OnClickListener() {
      @Override
      public void onClick(View v) {
        stopwatch.start();

//        new Handler().postDelayed(new Runnable() {
//          @Override
//          public void run() {
//            stopwatch.changeFormatWhen(20, TimeUnits.SECONDS, "Format20: SS -> LL#L");
//          }
//        }, 19000);
      }
    });

    findViewById(R.id.btn_stop).setOnClickListener(new OnClickListener() {
      @Override
      public void onClick(View v) {
        stopwatch.stop();
        Toast.makeText(MainActivity.this, "time = " + stopwatch.getTime(),
            Toast.LENGTH_SHORT).show();
      }
    });

    findViewById(R.id.btn_reset).setOnClickListener(new OnClickListener() {
      @Override
      public void onClick(View v) {
        stopwatch.reset();
        textTimer.setText("0");
      }
    });
  }
}

File: timerx-example/src/main/java/com/arsvechkarev/timerxexample/MainActivity.kt
package com.arsvechkarev.timerxexample

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter
import androidx.viewpager.widget.ViewPager
import com.google.android.material.tabs.TabLayout

class MainActivity : AppCompatActivity() {
  
  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    setContentView(R.layout.activity_main)
    val sectionsPagerAdapter = SectionsPagerAdapter(
      supportFragmentManager)
    val viewPager = findViewById<ViewPager>(R.id.view_pager)
    viewPager.adapter = sectionsPagerAdapter
    val tabs = findViewById<TabLayout>(R.id.tabs)
    tabs.setupWithViewPager(viewPager)
  }
  
  private inner class SectionsPagerAdapter(fm: FragmentManager) :
    FragmentPagerAdapter(fm, BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT) {
  
    override fun getItem(position: Int): Fragment {
      return if (position == 0)
        StopwatchFragment()
      else
        TimerFragment()
    }
  
    override fun getPageTitle(position: Int): CharSequence? {
      return if (position == 0)
        "Stopwatch"
      else
        "Timer"
    }
    
    override fun getCount(): Int {
      return 2
    }
  }
}
